src/viur/core/skeleton.py

```diff
@@ -1132,8 +1173,11 @@ def txnUpdate(write_skel):
             db_obj = skel.preProcessSerializedData(db_obj)
 
             # Allow the custom DB Adapter to apply last minute changes to the object
-            if skel.customDatabaseAdapter:
-                db_obj = skel.customDatabaseAdapter.preprocessEntry(db_obj, skel, change_list, is_add)
+            for adapter in utils.ensure_iterable(skel.customDatabaseAdapter):
+                db_obj = adapter.preprocessEntry(db_obj, skel, change_list, is_add)  # FIXME: DEPRECATED.
```

Comment:
Why should the `#FIXME: DEPRECATED` mean?

Reply:
Because all these events with ugly names like `preprocessEntry` or `updateEntry` are being superseded by the new `trigger` handler.

Reply:
But i find it clearer to directly subclass suitable methods for corresponding events than to build cases in a method.

So instead of 
```python
class MyAdapter(CustomDatabaseAdapter):
    def updateEntry(self, entity: db.Entity, skel: SkeletonInstance,
                    changeList: list[str], isAdd: bool) -> None:
        # my update logic

    def deleteEntry(self, entity: db.Entity, skel: SkeletonInstance) -> None:
        # my delete logic
``` 

I should do in future

```python
class MyAdapter(CustomDatabaseAdapter):
    def trigger(
        self,
        action: str,
        old_skel: db.Entity | SkeletonInstance = None,
        new_skel: SkeletonInstance = None,
        change_list: t.Iterable[str] = (),
        descr: t.Optional[str] = None,
        user: t.Optional[SkeletonInstance] = None,
        tags: t.Iterable[str] = (),
        db_write: bool = False
    ) -> str | None:
        match action:
            case "edit":
                # my update logic
        
            case "delete":
                # my delete logic
``` 

?

Reply:
Well, yes this is how its currently intended to be used.

The problem is similar to the ActionSkel event handling we discussed, and where I also don't have the perfect solution right now.

I would like to abandon ugly functions with ugly signatures defined for every operation, when it can be done  with just one function and conditional code on demand.